[macOS] Add a flag to keep track of when the web view is entering or exiting window fullscreen

whsieh opened the pull request:
#### 6097e4af324392c511a96affb39de098eb15182b
<pre>
[macOS] Add a flag to keep track of when the web view is entering or exiting window fullscreen
<a href="https://bugs.webkit.org/show_bug.cgi?id=294163">https://bugs.webkit.org/show_bug.cgi?id=294163</a>
<a href="https://rdar.apple.com/152741612">rdar://152741612</a>

Reviewed by Abrar Rahman Protyasha.

Add a flag, `m_windowIsEnteringOrExitingFullScreen`, that tracks when the web view&apos;s containing
window is animating to or from window fullscreen. See the paired change for more details.

* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(-[WKWindowVisibilityObserver startObserving:]):
(-[WKWindowVisibilityObserver stopObserving]):
(-[WKWindowVisibilityObserver _windowDidEnterOrExitFullScreen:]):
(-[WKWindowVisibilityObserver _windowWillEnterOrExitFullScreen:]):
(WebKit::WebViewImpl::windowWillEnterOrExitFullScreen):
(WebKit::WebViewImpl::windowDidEnterOrExitFullScreen):

Make sure we update content inset fill views after finishing the window fullscreen transition.

(WebKit::WebViewImpl::viewWillMoveToWindowImpl):

If the window containing the web view changes, make sure the flag is reset.

Canonical link: <a href="https://commits.webkit.org/295963@main">https://commits.webkit.org/295963@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3f5f597240736b602fcb20ebe2606487b978ceda

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106750 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26501 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16898 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111961 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/57348 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27171 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35003 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81049 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109754 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21561 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96326 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61390 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21000 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56802 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90901 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14462 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114891 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33888 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/24984 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90114 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34253 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92557 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89824 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22921 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34763 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12572 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/29536 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33813 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/39226 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33559 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36912 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35158 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3f5f597240736b602fcb20ebe2606487b978ceda. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46461

whsieh commented:
Thanks for the review!

whsieh committed:
[macOS] Add a flag to keep track of when the web view is entering or exiting window fullscreen
https://bugs.webkit.org/show_bug.cgi?id=294163
rdar://152741612

Reviewed by Abrar Rahman Protyasha.

Add a flag, `m_windowIsEnteringOrExitingFullScreen`, that tracks when the web view's containing
window is animating to or from window fullscreen. See the paired change for more details.

* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(-[WKWindowVisibilityObserver startObserving:]):
(-[WKWindowVisibilityObserver stopObserving]):
(-[WKWindowVisibilityObserver _windowDidEnterOrExitFullScreen:]):
(-[WKWindowVisibilityObserver _windowWillEnterOrExitFullScreen:]):
(WebKit::WebViewImpl::windowWillEnterOrExitFullScreen):
(WebKit::WebViewImpl::windowDidEnterOrExitFullScreen):

Make sure we update content inset fill views after finishing the window fullscreen transition.

(WebKit::WebViewImpl::viewWillMoveToWindowImpl):

If the window containing the web view changes, make sure the flag is reset.

Canonical link: https://commits.webkit.org/295963@main
